contracts/src/rollup/RollupUserLogic.sol

```diff
@@ -219,26 +219,37 @@ contract RollupUserLogic is RollupCore, UUPSNotUpgradeable, IRollupUser {
     /**
      * @notice Refund a staker that is currently staked on an assertion that either has a chlid assertion or is the latest confirmed assertion.
      */
-    function returnOldDeposit() external override onlyValidator whenNotPaused {
+    function returnOldDeposit() external override onlyValidator(msg.sender) whenNotPaused {
         requireInactiveStaker(msg.sender);
         withdrawStaker(msg.sender);
     }
 
+    /**
+     * @notice From the staker's withdrawal address, 
+     * refund a staker that is currently staked on an assertion that either has a chlid assertion or is the latest confirmed assertion.
+     */
+    function returnOldDepositFromWithdrawalAddress(address stakerAddress) external override onlyValidator(stakerAddress) whenNotPaused {
+        require(msg.sender == withdrawalAddress(stakerAddress), "NOT_WITHDRAWAL_ADDRESS");
+        requireInactiveStaker(stakerAddress);
+        withdrawStaker(stakerAddress);
```

Comment:
make these 2 lines a internal function and share it with `returnOldDeposit`

Reply:
ed8d9cff8d52d6744761285eaf0e1aca8f3602a6